launcher/ui/pages/modplatform/ModPage.cpp

```diff
@@ -241,6 +245,79 @@ void ModPage::onModSelected()
     ui->packView->adjustSize();
 }
 
+void ModPage::openUrl(const QUrl& url)
+{
+    // do not allow other url schemes for security reasons
+    if (!(url.scheme() == "http" || url.scheme() == "https")) {
+        qWarning() << "Unsupported scheme" << url.scheme();
+        return;
+    }
+
+    // detect mod URLs and search instead
+    int prefixLength = 0;
+    const char* page;
+
+    if ((url.host() == "modrinth.com" || url.host() == "www.modrinth.com") && url.path().startsWith("/mod/")) {
+        prefixLength = 5;
+        page = "modrinth";
+    } else if (APPLICATION->capabilities() & Application::SupportsFlame &&
+               (url.host() == "curseforge.com" || url.host().endsWith(".curseforge.com"))) {
+        if (url.path().toLower().startsWith("/minecraft/mc-mods/"))
+            prefixLength = 19;
+        else if (url.path().toLower().startsWith("/projects/"))
+            prefixLength = 10;
+        page = "curseforge";
+    }
```

Comment:
I think I would prefer just going with a RegEx instead of manually parsing it like this.

Extracting slug from a Modrinth link: `https?:\/\/(?:www\.)?modrinth\.com\/mod\/(\w+)\/?`
Extracing slug from CF: `https?:\/\/(?:www\.)?curseforge\.com\/minecraft\/mc-mods\/(\w+)\/?`
Extracing slug from CF: `https?:\/\/(?:www\.)?curseforge\.com\/projects\/(\w+)\/?`

You can use [`QRegularExpression::anchoredPattern`](https://doc.qt.io/qt-6/qregularexpression.html#anchoredPattern) to make the patterns match the full URL.